### ISSUE TITLE
[Flang] %REF: Incorrect diagnostic.

### ISSUE THREAD

**Issue by DanielCChen** [llvm/llvm-project#93489]
Consider the following code:
```
      interface
         subroutine sub(ctx)
            integer :: ctx
         end
      end interface

   integer, parameter :: ret = 5

   call sub(%ref(ret))

  end
  ```
  
  Flang currently issues an error as:
  ```
9:18: error: Assignment to constant 'ret' is not allowed
     call sub(%ref(ret))
                   ^^^
```

XLF, gfortran and ifort all accept the above syntax.

**Comment by klausler** [comment]
https://github.com/llvm/llvm-project/pull/93718
